File: stepik-java-api/src/main/java/org/stepik/api/objects/lessons/Lessons.java
package org.stepik.api.objects.lessons;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.objects.ObjectsContainer;

import java.util.ArrayList;
import java.util.List;

/**
 * @author meanmail
 */
public class Lessons extends ObjectsContainer<Lesson> {
    private List<Lesson> lessons;

    @NotNull
    public List<Lesson> getLessons() {
        if (lessons == null) {
            lessons = new ArrayList<>();
        }
        return lessons;
    }

    @NotNull
    @Override
    public List<Lesson> getItems() {
        return getLessons();
    }

    @NotNull
    @Override
    public Class<Lesson> getItemClass() {
        return Lesson.class;
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/lessons/Lessons.kt
package org.stepik.api.objects.lessons

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import org.stepik.api.client.serialization.DefaultAsEmptyLessonArrayAdapter
import org.stepik.api.objects.ObjectsContainer


class Lessons : ObjectsContainer<Lesson>() {
    @SerializedName("lessons")
    @JsonAdapter(DefaultAsEmptyLessonArrayAdapter::class, nullSafe = false)
    override var items = mutableListOf<Lesson>()

    override val itemClass
        get() = Lesson::class.java
}
